subsys/bluetooth/mesh/friend.c

```diff
@@ -164,6 +164,11 @@ static void friend_clear(struct bt_mesh_friend *frnd)
 	/* If cancelling the timer fails, we'll exit early in the work handler. */
 	(void)k_work_cancel_delayable(&frnd->timer);
 
+	for (int i = 0; i < ARRAY_SIZE(frnd->cred); i++) {
+		if (frnd->subnet->keys[i].valid) {
+			bt_mesh_friend_cred_destroy(&frnd->cred[i]);
```

Comment:
It isn't required. I didn't find any place where both credentials or keys were checked they have zeroes. There is `valid` flag that keeps information about the validity of credentials.  In case of the creation of the new credentials, the old key\key ID will be rewritten by the new one.
The same for LPN.